### PR TITLE
RDoc-2482 Python demos (Multi-Map Indexes + JS Indexes + Auto Indexes)

### PR DESCRIPTION
**Pull request by Danielle9897**
**Related issue:**
https://issues.hibernatingrhinos.com/issue/RDoc-2482/Python-Demo-examples

---

@poissoncorp  This PR has the following demos:

**Multi-Map indexes category:**
Multi-Map index - basic
Multi-Map index - customized fields
Multi-Map-Reduce index

**JS indexes category:**
JS map index

**Auto indexes category:**
Auto map index I
Auto map index II
Auto map-reduce index

### CODE REVIEW

File: DemoServer/AdditionalLanguages/python/multiMapIndexes/multiMapIndexBasic/metadata.json [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                "* Use `add_map()` to define the content of the index-field per collection indexed.<br>",
+                "  Data from the different collections is indexed into the _common index-field_ in the IndexEntry class.",
+                "",
+                "* In this example, document properties from Companies & Suppliers collections are indexed into the common index field _'name'_.<br>",
+                "  The document properties indexed are not customized but taken as is.<br>",
+                "  The document properties are case-sensitive, use the exact casing as in the document.",
+                "",
+                "* The index will contain a mapping from any term generated for the 'name' index-field to the relevant document in either the Companies or the Suppliers collection.",
+                "",
+                "* When querying this index, the index-field 'name' can be used to filter the query results and get matching documents from either collection."
```
Comment (poissoncorp) [inline DemoServer/AdditionalLanguages/python/multiMapIndexes/multiMapIndexBasic/metadata.json:94]: let's also mention `self._index("name", FieldIndexing.EXACT)` explicitly in this step

File: DemoServer/AdditionalLanguages/python/multiMapIndexes/multiMapIndexCustomizedFields/metadata.json [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            "Title": "Store index fields",
+            "Slug" : "step-5",
+            "DescriptionLines": [
+                "* Usually, when querying an index, the index provides the matching document ID, and results are taken from that document in the documents store.",
+                "",
+                "* _If the index-fields are customized_, and you want the query to return their content,<br>",
+                "  then these fields must be stored in the index.<br>",
+                "  The results will be taken directly from the index fields instead of from the document.",
+                "",
+                "* Call `store()` with param _FieldStorage.YES_ to store the data on the index."
```
Comment (poissoncorp) [inline DemoServer/AdditionalLanguages/python/multiMapIndexes/multiMapIndexCustomizedFields/metadata.json:155]: https://github.com/ravendb/demo/pull/193/files#r1350181511
it's a good line of documentation, that we might lack in the place I've annotated before

File: DemoServer/AdditionalLanguages/python/multiMapIndexes/multiMapIndexBasic/metadata.json (hunk 1 above)
Comment (poissoncorp) [reply DemoServer/AdditionalLanguages/python/multiMapIndexes/multiMapIndexBasic/metadata.json:94]: users may get confused what `index()` method does

**Review by poissoncorp — APPROVED** [review]

File: DemoServer/AdditionalLanguages/python/multiMapIndexes/multiMapIndexBasic/metadata.json (hunk 1 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/python/multiMapIndexes/multiMapIndexBasic/metadata.json:94]: @poissoncorp 

TBH - I think the following code line should Not be in this place in this demo
`self._index("name", FieldIndexing.EXACT)`

It is Not there originally for all the other languages of this demo, 
as it is not directly needed for the multi-map index explanation.

**Pull request merged**